src/tools/SwitchingFunction.h

```diff
@@ -39,85 +40,88 @@ class Keywords;
-  double do_rational(double rdist,double&dfunc,int nn,int mm)const;
+    double do_rational(double rdist,double&dfunc,int nn,int mm)const;
 /// Function for lepton;
-  std::string lepton_func;
+    std::string lepton_func;
 /// Lepton expression.
 /// \warning Since lepton::CompiledExpression is mutable, a vector is necessary for multithreading!
-  std::vector<lepton::CompiledExpression> expression;
+    std::vector<lepton::CompiledExpression> expression;
 /// Lepton expression for derivative
 /// \warning Since lepton::CompiledExpression is mutable, a vector is necessary for multithreading!
-  std::vector<lepton::CompiledExpression> expression_deriv;
-  std::vector<double*> lepton_ref;
-  std::vector<double*> lepton_ref_deriv;
+    std::vector<lepton::CompiledExpression> expression_deriv;
+    std::vector<double*> lepton_ref;
+    std::vector<double*> lepton_ref_deriv;
 /// Set to true for fast rational functions (depending on x**2 only)
-  bool fastrational=false;
+    bool fastrational=false;
 /// Set to true if lepton only uses x2
-  bool leptonx2=false;
+    bool leptonx2=false;
 public:
-  static void registerKeywords( Keywords& keys );
+    static void registerKeywords( Keywords& keys );
 /// Set a "rational" switching function.
 /// Notice that a d_max is set automatically to a value such that
 /// f(d_max)=0.00001.
-  void set(int nn,int mm,double r_0,double d_0);
+    void set(int nn,int mm,double r_0,double d_0);
 /// Set an arbitrary switching function.
 /// Parse the string in definition and possibly returns errors
 /// in the errormsg string
-  void set(const std::string& definition, std::string& errormsg);
+    void set(const std::string& definition, std::string& errormsg);
 /// Returns a string with a description of the switching function
-  std::string description() const ;
+    std::string description() const ;
 /// Compute the switching function.
 /// Returns s(x). df will be set to the value of the derivative
 /// of the switching function _divided_by_x
-  double calculate(double x,double&df)const;
+    double calculate(double x,double&df)const;
 /// Compute the switching function.
 /// Returns \f$ s(\sqrt{x})\f$ .
 /// df will be set to the \f$ \frac{1}{\sqrt{x}}\frac{ds}{d\sqrt{x}}= 2 \frac{ds}{dx}\f$
 /// (same as calculate()).
 /// The advantage is that in some case the expensive square root can be avoided
 /// (namely for rational functions, if nn and mm are even and d0 is zero)
-  double calculateSqr(double distance2,double&dfunc)const;
+    double calculateSqr(double distance2,double&dfunc)const;
 /// Returns d0
-  double get_d0() const;
+    double get_d0() const;
 /// Returns r0
-  double get_r0() const;
+    double get_r0() const;
 /// Return dmax
-  double get_dmax() const;
+    double get_dmax() const;
 /// Return dmax squared
-  double get_dmax2() const;
+    double get_dmax2() const;
+/// Set a r0 switching function.
+    void set_r0(double r_0);
```

Comment:
@SilvioPipolo notice that the switching function also stores `double invr0_2`. In addition, there are many other members that depend indirectly on `r0` (e.g. `stretch` and `shift`). I would recommend not allowing the parameters to be changed individually as you are doing.

Can you change the `PIV` class so that instead of setting `r0` you reset the whole switching function?

Giovanni
`

Reply:
@SilvioPipolo It also looks from github that you modified the whole file. Please can you add only the new change? You should use the correct version of astyle and avoid using some additional tool for reformatting.

Reply:
ok for the switching function. I'll try to do that. 
Concerning the files, I formatted them using astyle version
Artistic Style Version 3.1
on linux because of the errors I got from travis